src/PhoneNumberInput/PhoneNumberInput.jsx

```diff
@@ -116,15 +132,32 @@ const PhoneNumberInput = forwardRef((props, ref) => {
     onCountryNumberChange(country)
   }
 
+  const onCursorPositionChanged = () => {
+    setCursorPositionChangedToggler(!cursorPositionChangedToggler)
+  }
+
   const onCountryNumberChange = country => {
-    const value = `${country.prefix}${displayValue}`
+    const value = `${country.prefix}${trimStartingZero(displayValue)}`
     const event = { target: { value } }
     onChange(event)
   }
+
   const onNumberChange = event => {
-    event.target.value = `${country.prefix}${event.target.value}`
+    if (trimStartingZero(event.target.value) !== event.target.value) {
+      setCursorPosition(0)
+    } else {
+      setCursorPosition(event.target.selectionStart)
+    }
```

Comment:
The cursor should already be at the selectionStart position. Are you sure this is not a noop?
```suggestion
    }
```